src/sage/schemes/elliptic_curves/gal_reps_number_field.py

```diff
@@ -402,6 +402,369 @@ def reducible_primes(self):
 
         return [l for l in self.isogeny_bound() if self.E.isogenies_prime_degree(l)]
 
+    def is_borel(self, p):
+        r"""
+        Return ``True`` if the projective mod-`p` image is contained in a Borel subgroup.
+
+        INPUT:
+
+        - ``p`` (integer) -- a prime number
+
+        OUTPUT:
+
+        (boolean) ``True or ``False``.
+
+        EXAMPLES::
+
+            sage: from sage.schemes.elliptic_curves.gal_reps_number_field import GaloisRepresentation
+            sage: E = EllipticCurve('56b1')
+            sage: G = GaloisRepresentation(E)
+            sage: G.is_borel(2)
+            True
+            sage: E = EllipticCurve('50a1')
+            sage: G = GaloisRepresentation(E)
+            sage: G.is_borel(3)
+            True
+            sage: G.is_borel(5)
+            True
+            sage: E.has_cm()
+            False
+            sage: E = EllipticCurve('405d1')
+            sage: G = GaloisRepresentation(E)
+            sage: G.is_borel(7)
+            True
+            sage: E = EllipticCurve('121a1')
+            sage: G = GaloisRepresentation(E)
+            sage: G.is_borel(11)
+            True
+            sage: E = EllipticCurve('147c1')
+            sage: G = GaloisRepresentation(E)
+            sage: G.is_borel(13)
+            True
+
+        TESTS::
+
+            sage: from sage.schemes.elliptic_curves.gal_reps_number_field import GaloisRepresentation
+            sage: E = EllipticCurve('11a1')
+            sage: G = GaloisRepresentation(E)
+            sage: G.is_borel(4)
+            Traceback (most recent call last):
+            ...
+            NotImplementedError: p = 4 should be prime
+            sage: G.is_borel(x)
+            Traceback (most recent call last):
+            ...
+            ValueError: p = x should be a prime integer
+        """
+        if p == 2:
+            return not self.E.division_polynomial(2).is_irreducible()
+
+        # for p = 3, 5, 7, 13 we use the fact that X_0(p) has genus 0
+        if p in [3, 5, 7, 13]:
+            from sage.schemes.elliptic_curves.isogeny_small_degree import Fricke_polynomial
+            j = self.E.j_invariant()
+            x = polygen(j.parent())
+            return bool((Fricke_polynomial(p)(x)-j*x).roots()) # True iff list not empty
+
+        try:
+            p = ZZ(p)
+        except TypeError:
+            raise ValueError("p = {} should be a prime integer".format(p))
```

Comment:
```suggestion
            raise ValueError(f"{p = } should be a prime integer")
```

Do you know about f-strings? They are very useful (https://docs.python.org/3/tutorial/inputoutput.html#tut-f-strings)

Reply:
I will look these up and use them in future, but do not feel inclined to change all these strings now.